core/src/toga/app.py

```diff
@@ -778,18 +805,19 @@ def startup(self) -> None:
         Subclasses can override this method to define customized startup behavior.
         """
 
-    def _open(self, path):
+    def _open(self, path: Path) -> None:
         """Internal utility method; open a new document in this app, and shows the document.
 
         :param path: The path to the document to be opened.
         :raises ValueError: If the document is of a type that can't be opened. Backends can
-            suppress this exception if necessary to presere platform-native behavior.
+            suppress this exception if necessary to preserve platform-native behavior.
         """
         try:
             DocType = self.document_types[path.suffix[1:]]
         except KeyError:
             raise ValueError(f"Don't know how to open documents of type {path.suffix}")
         else:
-            document = DocType(path, app=self)
+            # TODO:PR: does this need `document_type`? or is `Document` not the right type?
+            document = DocType(path, app=self)  # type: ignore[call-arg]
             self._documents.append(document)
             document.show()
```

Comment:
Trying to figure out which right here...the typing or the code.

Does `DocType()` need the `document_type` specified as a argument?

Reply:
Which is right? Probably neither :-P

I wouldn't get too invested in the typing on Document/DocumentApp - the API changes a bit in #2244. 

However, in this incarnation, DocType is a subclass of Document; and one of the requirements of the subclass is that the `document_type` argument is populated by the constructor.